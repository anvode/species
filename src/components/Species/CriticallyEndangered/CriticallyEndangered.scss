.critically-endangered {
    &__title {
        text-align: center;
    }
}
