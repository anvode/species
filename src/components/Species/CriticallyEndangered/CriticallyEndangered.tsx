import * as React from 'react';
import { useContext } from 'react';

import { SpeciesContext } from '../../../context/species/SpeciesContext';

import './CriticallyEndangered.scss';

export interface CriticallyEndangeredProps {}

const CriticallyEndangered: React.FC<CriticallyEndangeredProps> = () => {
    const { state: {criticallyEndangered, region} } = useContext(SpeciesContext);

    return <div className="critically-endangered">
        <h2 className="critically-endangered__title">Critically Endangered Species in {region}</h2>
        {criticallyEndangered.length > 0 ? (
            <div className="species-results">
                <p>{criticallyEndangered.length} Species found</p>
                {criticallyEndangered.map(item => (
                    <div key={item.taxonid} className="species-results__item">
                        <div className={'species-results__name'}>{item.scientific_name}</div>
                        <div className="species-results__value">{item.measures && item.measures.length > 0 ? item.measures: 'empty field'}</div>
                    </div>
                ))}
            </div>
        ): 'No results'}
    </div>;
};

export default CriticallyEndangered;