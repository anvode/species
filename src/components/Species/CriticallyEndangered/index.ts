import CriticallyEndangered from './CriticallyEndangered';

export default CriticallyEndangered;