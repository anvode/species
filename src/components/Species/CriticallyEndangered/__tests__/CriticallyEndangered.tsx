import * as React from 'react';
import { render, cleanup } from '@testing-library/react';
import CriticallyEndangered from '../CriticallyEndangered';

afterEach(cleanup);

describe('CriticallyEndangered Species', () => {
    it('snapshot CriticallyEndangered Species', () => {
        const { asFragment } = render(<CriticallyEndangered />);
        expect(asFragment()).toMatchSnapshot();
    });
});

