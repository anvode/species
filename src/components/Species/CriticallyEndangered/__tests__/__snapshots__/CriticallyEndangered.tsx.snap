// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CriticallyEndangered Species snapshot CriticallyEndangered Species 1`] = `
<DocumentFragment>
  <div
    class="critically-endangered"
  >
    <h2
      class="critically-endangered__title"
    >
      Critically Endangered Species in 
    </h2>
    No results
  </div>
</DocumentFragment>
`;
